wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -529,6 +531,47 @@ createHdWallet newRoot defaultHdAccountId defaultHdAddress utxoByAccount =
                Nothing ->
                    Map.insert defaultHdAccountId (mempty, [defaultAddr]) m
 
+-- | Create an HdWallet with HdRoot
+--
+-- NOTE: We allow an initial set of accounts with associated addresses and
+-- balances /ONLY/ for testing purpose. Normally this should be empty; see
+-- 'createHdWallet'/'createWalletHdRnd' in "Cardano.Wallet.Kernel.Wallets".
+--
+-- INVARIANT: Creating a new wallet always come with a fresh HdAccount and
+-- a fresh 'HdAddress' attached to it, so we have to pass these two extra
+-- piece of into to the update function. We do @not@ build these inside the
+-- update function because derivation requires an 'EncryptedSecretKey' and
+-- definitely we do not want it to show up in our acid-state logs.
+--
+createExternalHdWallet :: HdRoot
+                       -> HdAccountId
+                       -- ^ The default HdAccountId to go with this HdRoot. This
+                       -- function will take responsibility of creating the associated
+                       -- 'HdAccount'.
+                       -> Map HdAccountId (Utxo, [AddrWithId])
+                       -> Update DB (Either HD.CreateHdRootError ())
+createExternalHdWallet newRoot defaultHdAccountId utxoByAccount =
```

Comment:
Do external wallets share the same account id representation than the current wallets?